lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -838,80 +801,190 @@ spec = do
                 [ expectField #delegation (`shouldBe` notDelegating [])
                 ]
 
-    it "STAKE_POOLS_JOIN/QUIT - Checking delegation expectations with \
-       \multiple certs inserted in a given epoch" $ \ctx -> do
-        (_, p1:p2:p3:_) <- eventually "Stake pools are listed" $
+    it "STAKE_POOL_NEXT_02 - Override join with join in the same epoch =>\
+        \ delegating to the last one in the end" $ \ctx -> do
+        (_, p1:p2:_) <- eventually "Stake pools are listed" $
             unsafeRequest @[ApiStakePool] ctx Link.listStakePools Empty
         w <- fixtureWallet ctx
 
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty
+            >>= flip verify
+                [ expectField #delegation (`shouldBe` notDelegating []) ]
+
+        -- make sure we are at the beginning of new epoch
+        (currentEpoch, sp) <- getSlotParams ctx
+        waitForNextEpoch ctx
+
+        let joinPool pool = do
+                r1 <- joinStakePool ctx (pool ^. #id) (w, fixturePassphrase)
+                expectResponseCode HTTP.status202 r1
+                waitAllTxsInLedger ctx w
+                request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty >>= flip verify
+                    [ expectField #delegation
+                        (`shouldBe` notDelegating
+                            [ (Just (pool ^. #id), mkEpochInfo (currentEpoch + 3) sp)
+                            ]
+                        )
+                    ]
+
+        -- join p1 and p2 in the same epoch and make sure delegation next is
+        -- always to the last one joined
+        joinPool p1
+        joinPool p2
+
+    it "STAKE_POOL_NEXT_02 - Override join with quit in the same epoch =>\
+        \ not_delegating in the end" $ \ctx -> do
+        (_, pool:_) <- eventually "Stake pools are listed" $
+            unsafeRequest @[ApiStakePool] ctx Link.listStakePools Empty
+        w <- fixtureWallet ctx
+
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty
+            >>= flip verify
+                [ expectField #delegation (`shouldBe` notDelegating []) ]
+
+        -- make sure we are at the beginning of new epoch
+        (currentEpoch, sp) <- getSlotParams ctx
+        waitForNextEpoch ctx
+
+        -- join SP
+        r1 <- joinStakePool ctx (pool ^. #id) (w, fixturePassphrase)
```

Comment:
why not to move `joinPool` definition outside `STAKE_POOL_NEXT_02 - Override join with join in the same epoch =>delegating to the last one in the end` and use here and below?